client/ayon_core/pipeline/create/context.py

```diff
@@ -1987,12 +1987,16 @@ def create(
                     "Folder '{}' was not found".format(folder_path)
                 )
 
-        task_name = None
         if task_entity is None:
             task_name = self.get_current_task_name()
-            task_entity = ayon_api.get_task_by_name(
-                project_name, folder_entity["id"], task_name
-            )
+            if task_name:
+                task_entity = ayon_api.get_task_by_name(
+                    project_name, folder_entity["id"], task_name
+                )
```

Comment:
Why do we need the `if task_name` logic suddenly? Isn't there **always** a current task name?

Also, I wonder if it makes sense to rename the `task_name` variable here to `current_task_name` to avoid confusion with the `task_name` variable further down below.

Reply:
> Why do we need the if task_name logic suddenly? Isn't there always a current task name?

No there is not. E.g. in traypublisher is set only project.

> Also, I wonder if it makes sense to rename the task_name variable here to current_task_name to avoid confusion with the task_name variable further down below.

Yes, could be.